nibe/connection/nibegw.py

```diff
@@ -267,6 +316,21 @@ def _on_raw_coil_value(self, coil_address: int, raw_value: bytes):
         logger.info(f"{coil.name}: {coil.value}")
         self._heatpump.notify_coil_update(coil)
 
+    def _on_coil_value(self, coil_address: int, value: Union[float, int, str]):
+        try:
+            coil = self._heatpump.get_coil_by_address(coil_address)
+        except CoilNotFoundException:
+            if coil_address == 65535:  # 0xffff
+                return
+            raise
+
+        if coil.mappings and isinstance(value, int):
+            value = coil.mappings[str(value)]
```

Comment:
This is ugly. Would be better if we could write a non mapped value somehow.

Reply:
We'd better have all defined. I'd better see an error that a mapping is missing than silently writing something somewhere.

Reply:
my point is we are working with pre-mapped data here. So it would have been better to have a setter that allowed non mapped data.